server/server_test.go

```diff
@@ -0,0 +1,80 @@
+package main
+
+import (
+	"github.com/99designs/gqlgen/client"
+	"github.com/99designs/gqlgen/graphql/handler"
+
+	"go-gql-typescript-example/graph"
+	"go-gql-typescript-example/graph/generated"
+	"go-gql-typescript-example/graph/model"
+
+	"github.com/stretchr/testify/require"
+
+	"log"
+	"os"
+	"testing"
+)
+
+var user *model.User
+
+func TestMain(m *testing.M) {
+	graph.Database = openAndMigrateDatabase("file::memory:?cache=shared")
+
+	user = createUser("bob")
+
+	if user == nil {
+		log.Fatalf("failed to create default user")
+	}
+
+	os.Exit(m.Run())
+}
+
+func TestFetchTodos(t *testing.T) {
+	c := client.New(handler.NewDefaultServer(generated.NewExecutableSchema(generated.Config{Resolvers: &graph.Resolver{}})))
+
+	t.Run("Fetch user list", func(t *testing.T) {
+		// response struct must match the shape of the graphql JSON response
+		var resp struct {
+			Users []*model.User `json:"users"`
```

Comment:
is `json:"users"` needed?

Reply:
Absolutely yeah, this shape needs to match the response object for the query and runs through the graphql stack, including a json encode / decode.  It's definitely a bit of a pain to get right when writing new tests.